router_api.go

```diff
@@ -15,8 +31,17 @@ func newAPIHandler(rout *Router) http.Handler {
 			w.WriteHeader(http.StatusMethodNotAllowed)
 			return
 		}
-
-		rout.ReloadRoutes()
+		// Send a message to the reload goroutine, which will start a new timeout
+		// before reloading, or do nothing if one is already in progress.
+		select {
+		case reloadChan <- true:
+			logInfo("router: reload triggered")
+			w.WriteHeader(http.StatusAccepted)
+			w.Write([]byte("Reload triggered"))
+		default:
+			logInfo("router: reload already in progress")
+			w.Write([]byte("Reload already in progress"))
+		}
```

Comment:
This looks a little racy to me. I think there's a case here, where a reload request could come in after the reload has completed, but not actually trigger a reload.

I think we need some sort of sentinel that's only in place around the sleep, but reset before the actual reload starts so that any requests that some in after the reload has started will trigger another sleep and reload once it's complete.